examples/notebooks/load-modelling.ipynb

```diff
@@ -0,0 +1,502 @@
+    "average_loads = []\n",
+    "prices = []\n",
+    "for n in all_networks:\n",
+    "    average_loads.append(n.loads_t.p.mean().sum())\n",
+    "    prices.append((n.objective + n.objective_constant) / n.loads_t.p.sum().sum())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "first_demand_iteration = average_loads[0]\n",
+    "updated_average_loads = average_loads[\n",
+    "    1:\n",
+    "]  # demand value in first iteration didn't have a price\n",
+    "updated_prices = prices[:-1]  # last price value wasn't used\n",
+    "\n",
+    "# Scatter plot of the points\n",
+    "plt.scatter(updated_average_loads, updated_prices, label=\"Elastic demand\")\n",
+    "\n",
+    "# Add numbers to each point\n",
+    "for i, (x, y) in enumerate(zip(updated_average_loads, updated_prices)):\n",
+    "    plt.annotate(\n",
+    "        \"Iteration \" + str(i + 1),\n",
+    "        (x, y),\n",
+    "        textcoords=\"offset points\",\n",
+    "        xytext=(30, 5),\n",
+    "        ha=\"center\",\n",
+    "    )\n",
+    "\n",
+    "# Fit a linear curve to the data\n",
+    "slope, intercept = np.polyfit(updated_average_loads, updated_prices, 1)\n",
+    "\n",
+    "# Linear curve\n",
+    "x = np.linspace(min(updated_average_loads), max(updated_average_loads), 100)\n",
+    "y = slope * x + intercept\n",
+    "plt.plot(x, y, label=f\"Linear curve\", color=\"blue\")\n",
+    "\n",
+    "plt.xlabel(\"Demand (MWh)\")\n",
+    "plt.ylabel(\"Price (Euro/MWh)\")\n",
+    "plt.title(\"Elasticity Curve\")\n",
+    "plt.legend(loc=\"upper right\")\n",
+    "plt.xlim([0.15, 0.35])\n",
+    "plt.ylim([35, 60])\n",
+    "plt.plot(x, y)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can observe from the elastic demand the following:\n",
+    "- Our initial average demand estimate of 0.2 MWh lead to a price resulting in more demand -> **iteration 1**.\n",
+    "- Using the new demand derieved from the elasticity curve and **iteration 1**, increased the average system price -> **iteration 2**\n",
+    "- ...\n",
+    "Luckily, after several iterations we observe that the results are **converging.**"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Note.** Here we solve the elastic demand as linear programming (LP) problem and several iterations. It is also possible to solve the problem with **one single** optimization run. However, this requires to reformulate the problem as quadratic programming (QP) problem, see [here](https://github.com/PyPSA/PyPSA/issues/574#issuecomment-1451865672). PyPSA can also do that, however, we need to add the QP solving capability to Linopy. This is already tracked as [feature request](https://github.com/PyPSA/linopy/issues/18). Stay tuned!"
```

Comment:
Needs to be adjusted. Linopy has now capabilities to solve quadratic problems (QP)

Reply:
@AnasAlgarei , regarding the smart thermostat example.
I think we should also dismiss the "smart thermostat" example. A thermostat is a measurement device. People find it irritating when a thermostat is modelled as a storage unit without a detailed explanation. Can you find another load-shifting example that is more focused ideally on electricity? We need a short and concise description of how the device behaves and why its modelled like e.g. a storage_unit

Further, I think the input:
- `carrier` could be removed. 
-`p_nom` could be set positive to say that there is a device.
-`cycling losses` (you know which variable I mean) could be removed. It increases unnecessary the complexity.

Reply:
> Needs to be adjusted. Linopy has now capabilities to solve quadratic problems (QP)

Apologies about this, realised I didn't save the last changes before committing, all sorted now.

Reply:
Hi @pz-max, I made changes as per your comments. Please review when possible.